docs/courses/robotics/3.5-computer-vision.md

```diff
@@ -0,0 +1,157 @@
+
+(3.5-orientation)=
+# 🧩 3.5 Computer Vision
+
+```{contents}
+:depth: 3
+```
+
+## 🔰 Tutorial
+
+In this module, you will develop software to:
+1. Demonstrate spatial referencing and ID lookup by using **OpenCV** and **AprilTags**
+2. Use a motorized microscope and **OpenCV** to search for regions of interest (ROI) in a sample
+
+### OpenCV
+
+First, you will utilize **OpenCV**, a powerful computer vision library, to preprocess images, detect objects, and perform real-time computer vision tasks. **OpenCV** is widely used for various computer vision applications, including image filtering, object tracking, and edge detection.
+
+#### Bill of Materials
+
+- [OpenCV](https://pypi.org/project/opencv-python/) (for Python installation: `pip install opencv-python`)
+- [AprilTag Python Library](https://pypi.org/project/apriltag/)
+- [Motorized Microscope]
+- [Raspberry Pi Camera Module](https://www.raspberrypi.org/products/camera-module-v2/)
+- [USB-A to micro USB-B cable]
+
+#### Demo
+
+✅ Read the [OpenCV Documentation](https://docs.opencv.org/4.x/)
+
+✅ Watch [Introduction to OpenCV](https://www.youtube.com/watch?v=oXlwWbU8l2o)
+
+In this task, you will use **OpenCV** to perform image preprocessing, such as applying a Gaussian blur, detecting edges, and identifying regions of interest (ROI).
+
+```python
+import cv2
+import numpy as np
+
+# Load an image
+img = cv2.imread('sample_image.jpg')
+
+# Apply Gaussian Blur
+blurred = cv2.GaussianBlur(img, (5, 5), 0)
+
+# Edge Detection
+edges = cv2.Canny(blurred, 100, 200)
+
+cv2.imshow('Edges', edges)
+cv2.waitKey(0)
+cv2.destroyAllWindows()
+```
+
+The above code will help you preprocess the image to smoothen it and then detect edges. This is the first step before applying any region of interest search.
+
+---
+
+### AprilTags
+
+Next, you will utilize **AprilTags**, a popular fiducial marker system for reliable and robust spatial referencing and object identification. AprilTags are often used in robotics for locating objects and navigating environments.
+
+#### Bill of Materials
+
+- [AprilTags Python Library](https://pypi.org/project/apriltag/)
+- [AprilTags Documentation](https://april.eecs.umich.edu/software/apriltag.html)
+- [Raspberry Pi Camera Module](https://www.raspberrypi.org/products/camera-module-v2/)
+- [Motorized Microscope]
+#### Demo
+
+✅ Watch [Vision Programming with AprilTags](https://youtu.be/TG9KAa2EGzQ)
+
+The following code demonstrates how to use **AprilTags** for spatial referencing by detecting tags and performing ID lookup.
```

Comment:
Would be good to have something demonstrating spatial orientation/locating based on the AprilTag.